src/components/Auth/types.ts

```diff
@@ -48,6 +48,8 @@ export const providerToIconMap = {
   'yahoo.com': 'assets/provider-icons/auth_service_yahoo.svg',
   phone: 'assets/provider-icons/auth_service_phone.svg',
   password: 'assets/provider-icons/auth_service_email.svg',
+  oidc: 'assets/provider-icons/auth_service_oidc.svg',
+  saml: 'assets/provider-icons/auth_service_saml.svg',
 };
 
 export interface AuthProviderInfo {
```

Comment:
If you update the type of `providerId` here to:

```ts
export interface AuthProviderInfo {
  providerId: Exclude<keyof typeof providerToIconMap, 'oidc' | 'saml'> | `oidc.${string}` | `saml.${string}`;
}
```

Then you don't need to use `any` in the tests ([here](https://github.com/firebase/firebase-tools-ui/blob/e985f23c3436553365782f218119b080f4412ed6/src/components/Auth/UsersCard/table/ProviderCell.test.tsx#L80-L81)).

[example in the TS playground](https://www.typescriptlang.org/play?#code/MYewdgzgLgBADgJxANwJYBMCmCAqICSoYAsgIZwwC8MA3gFAyPykQQDuIC6AXDAOSk+AGgZMQGYLz4AjYaMYRSAWwA2U4HzoBfOnUwAPOJ1iowUbADNSwTDACCAVygALAApI0WBPjAWQtUUQUDGx8HhgAUX1gFQcsAB4Aa0wATxALGCgUuEx0+A8Q3AIiMjghfnF0DRgAH35FVT4APlqYAANK4AA6ABIaaARTAHMtNta2hpVe-qhBsBG2gG5tXSJoZlYOLndgr15HFx3PUN9-anomfN3Q8L44FnZOdE0dNdhJo8L9pzcCrx8-FQApcgsdvLdJl1zNAXroAPRwmBsWxsUhmTLOVAQTL+bBIBAwEBOGDSTDAUgOCC2ABWlNgACJJvSYKQVOBbFiWWAYKZkKyMFcwXQ3jywHyVBgAMrKFSfPb2H5yk6A87yQWFMJSSYvIA)

Reply:
\* mindblown * whoa, I didn't know that you could do a pattern matched string type! I think I'll leave it as is though because this moves the casting to ProviderCell.tsx, thank you regardless!